core/spec/models/spree/line_item_spec.rb

```diff
@@ -220,10 +220,32 @@ def copy_price
     context 'when the price has a currency different from the order currency' do
       let(:currency) { "RUB" }
 
-      it 'raises an exception' do
-        expect {
+      before do
+        stub_spree_preferences(raise_with_invalid_currency: raise_with_invalid_currency)
+      end
+
+      context 'when raise_with_invalid_currency preference is true' do
+        let(:raise_with_invalid_currency) { true }
+
+        it 'raises an exception' do
+          expect {
+            line_item.money_price = new_price
+          }.to raise_exception(
+            Spree::LineItem::CurrencyMismatch,
+            'Line item price currency must match order currency!'
```

Comment:
I added this check because I changed the way the error message is retrieved and I wanted to have a real check on that. Now that I know it's working I can remove it if you think it's not needed.

Reply:
If you want to check that the error message is retrieved correctly, then you should definitely keep this in place as that logic may break for some reason after the original PR is merged.